src/runtime/http.rs

```diff
@@ -54,28 +69,50 @@ impl HttpClient {
         &self,
         method: Method,
         uri: &str,
-        headers: impl IntoIterator<Item = &'a (&'a str, &'a str)>,
-    ) -> reqwest::Result<String> {
-        let text = self.request(method, uri, headers).await?.text().await?;
+        headers: &'a [(&'a str, &'a str)],
+    ) -> Result<String, Box<dyn Error>> {
+        let res = self.request(method, Uri::from_str(uri)?, headers).await?;
+
+        let mut buf = body::aggregate(res.into_body()).await?;
+        let mut bytes = vec![0; buf.remaining()];
+        buf.copy_to_slice(&mut bytes);
 
+        let text = String::from_utf8(bytes)?;
         Ok(text)
     }
 
-    /// Executes an HTTP equest and returns the response.
-    pub(crate) async fn request<'a>(
-        &self,
+    #[allow(clippy::type_complexity)]
+    /// Executes an HTTP request and returns the response.
+    pub(crate) fn request<'a>(
+        &'a self,
         method: Method,
-        uri: &str,
-        headers: impl IntoIterator<Item = &'a (&'a str, &'a str)>,
-    ) -> reqwest::Result<Response> {
-        let response = headers
-            .into_iter()
-            .fold(self.inner.request(method, uri), |request, (k, v)| {
-                request.header(*k, *v)
-            })
-            .send()
-            .await?;
+        uri: Uri,
+        headers: &'a [(&'a str, &'a str)],
+    ) -> Pin<Box<dyn Future<Output = Result<Response<Body>, Box<dyn Error>>> + 'a + Send>> {
```

Comment:
Can we write this type as `BoxFuture<'a, Result<Response<Body>, Box<dyn Error>>>` instead? It's a little shorter / easier to read. This may allow us to drop the clippy suppression too.